lib/live_content_report_exporter.rb

```diff
@@ -0,0 +1,40 @@
+require 'csv'
+
+class LiveContentReportExporter
+  attr_reader :publishing_apps
+
+  def initialize(publishing_apps)
+    @publishing_apps = publishing_apps
+  end
+
+  def total
+    editions.count
+  end
+
+  def file_path
+    @file_path ||= Rails.root.join("live_content_report_#{publishing_apps.join('_')}_#{Time.zone.now.strftime('%Y%m%d%H%M%S%N')}.csv")
+  end
+
+  def export(progress: nil)
+    CSV.open(file_path, 'w') do |csv_out|
+      csv_out << ['URL', 'Page title', 'Format', 'First published at']
+      editions.find_each.with_index do |edition, i|
+        progress.call(i, total) unless progress.nil?
+        csv_out << [
```

Comment:
I might extract this into a `csv_columns_for_row(edition)` method and then have a separate `csv_headers` method which returns the first row headers.

Reply:
yeah, that gives us an extension point if we want other types of reports.  :+1: